src/puya/awst/nodes.py

```diff
@@ -786,23 +787,24 @@ class TupleItemExpression(Expression):
 
     Note: this is its own item (vs IndexExpression) for two reasons:
     1. It's not a valid lvalue (tuples are immutable)
-    2. The index must always be a literal
+    2. The index must always be a literal, and can be negative
     """
 
     base: Expression
-    index: int
+    index: int = attrs.field(validator=attrs.validators.ge(0))
```

Comment:
Doesn't this validator go against point 2. above?

Reply:
I see the eb is converting it to a positive index, so perhaps it's point 2. that needs rewording

Reply:
ah, this can actually be removed, wasn't required in the end